### PR TITLE
[tests] Document why mmap tests are disabled with Linux/SGX

### PR DESCRIPTION
**Pull request by oshogbo**
## Description of the changes 

While here, let's skip the tests instead of expecting failure for Linux/SGX.
This is small optimization for our CI.

Fixes https://github.com/gramineproject/gramine/issues/663.

## How to test this PR? 

Run CI with sgx and with direct.

- - -
This change is [Reviewable](https://reviewable.io/reviews/gramineproject/gramine/1306)

### CODE REVIEW

**Review by dimakuv — CHANGES_REQUESTED** [review]
> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/1306)* status: all files reviewed, 7 unresolved discussions, not enough approvals from maintainers (3 more required), not enough approvals from different teams (2 more required, approved so far: ) (waiting on @oshogbo)

___
*[`-- commits` line 2 at r1](https://reviewable.io/reviews/gramineproject/gramine/1306#-NTtQGD6BwFzAojVBpR4):*
`...why mmap FS tests ... disabled on Linux-SGX`
___
*[`-- commits` line 5 at r1](https://reviewable.io/reviews/gramineproject/gramine/1306#-NTtQSrKDU8xSI_Vi7ki):*
`Linux-SGX`

("Linux-SGX" is the official name of our PAL, and we are in the core codebase, so I don't see a reason to use "Linux/SGX" here).
___
*[`libos/test/fs/test_fs.py` line 289 at r1](https://reviewable.io/reviews/gramineproject/gramine/1306#-NTtQgNE8xyHaJ4wpzaz:-NTtQgNE8xyHaJ4wpzb-:b-ye4rv4) ([raw file](https://github.com/gramineproject/gramine/blob/c91546a4e60fe97d50cd455601f44d637be6e186/libos/test/fs/test_fs.py#L289)):*
> ```Python
>         self.do_copy_test('copy_sendfile', 60)
> 
>     # These tests are disabled for Linux/SGX because it doesn't implement mmap(PROTO_WRITE)
> ```

`Linux-SGX`
___
*[`libos/test/fs/test_fs.py` line 291 at r1](https://reviewable.io/reviews/gramineproject/gramine/1306#-NTtQ2Ia9Zj43FcITAhg:-NTtQ2Ib8EHzhDvIJArs:b-ciin34) ([raw file](https://github.com/gramineproject/gramine/blob/c91546a4e60fe97d50cd455601f44d637be6e186/libos/test/fs/test_fs.py#L291)):*
> ```Python
>     # These tests are disabled for Linux/SGX because it doesn't implement mmap(PROTO_WRITE)
>     # which is used in this test. We decided not to implement mmap(PROTO_WRITE) as we don't
>     # know any workload with it.
> ```

`any workload that uses it`
___
*[`libos/test/fs/test_fs.py` line 292 at r1](https://reviewable.io/reviews/gramineproject/gramine/1306#-NTtQ5y475m8oDanIeVy:-NTtQ5y475m8oDanIeVz:b-42oseh) ([raw file](https://github.com/gramineproject/gramine/blob/c91546a4e60fe97d50cd455601f44d637be6e186/libos/test/fs/test_fs.py#L292)):*
> ```Python
>     # which is used in this test. We decided not to implement mmap(PROTO_WRITE) as we don't
>     # know any workload with it.
>     @unittest.skipIf(HAS_SGX, 'mmap(PROTO_WRITE) not implemented')
> ```

`...not implemented in Linux-SGX PAL`
___
*[`libos/test/fs/test_fs.py` line 296 at r1](https://reviewable.io/reviews/gramineproject/gramine/1306#-NTtQA-iD725q471G5dI:-NTtQA-iD725q471G5dJ:b-7ropln) ([raw file](https://github.com/gramineproject/gramine/blob/c91546a4e60fe97d50cd455601f44d637be6e186/libos/test/fs/test_fs.py#L296)):*
> ```Python
>         self.do_copy_test('copy_mmap_whole', 30)
> 
>     @unittest.skipIf(HAS_SGX, 'mmap(PROTO_WRITE) not implemented')
> ```

ditto
___
*[`libos/test/fs/test_fs.py` line 300 at r1](https://reviewable.io/reviews/gramineproject/gramine/1306#-NTtQCCwEbG6S9q_kzof:-NTtQCCwEbG6S9q_kzog:b-7ropln) ([raw file](https://github.com/gramineproject/gramine/blob/c91546a4e60fe97d50cd455601f44d637be6e186/libos/test/fs/test_fs.py#L300)):*
> ```Python
>         self.do_copy_test('copy_mmap_seq', 60)
> 
>     @unittest.skipIf(HAS_SGX, 'mmap(PROTO_WRITE) not implemented')
> ```

ditto

**Review by oshogbo — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/1306)* status: all files reviewed, 7 unresolved discussions, not enough approvals from maintainers (3 more required), not enough approvals from different teams (2 more required, approved so far: ) (waiting on @dimakuv)

___
*[`-- commits` line 2 at r1](https://reviewable.io/reviews/gramineproject/gramine/1306#-NTtQGD6BwFzAojVBpR4):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

`...why mmap FS tests ... disabled on Linux-SGX`
</blockquote></details>

Done.
___
*[`-- commits` line 5 at r1](https://reviewable.io/reviews/gramineproject/gramine/1306#-NTtQSrKDU8xSI_Vi7ki):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

`Linux-SGX`

("Linux-SGX" is the official name of our PAL, and we are in the core codebase, so I don't see a reason to use "Linux/SGX" here).
</blockquote></details>

Done.
___
*[`libos/test/fs/test_fs.py` line 289 at r1](https://reviewable.io/reviews/gramineproject/gramine/1306#-NTtQgNE8xyHaJ4wpzaz:-NTtTjj5ElEJSp0seN8_:b-896fix) ([raw file](https://github.com/gramineproject/gramine/blob/c91546a4e60fe97d50cd455601f44d637be6e186/libos/test/fs/test_fs.py#L289)):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

`Linux-SGX`
</blockquote></details>

Done.
___
*[`libos/test/fs/test_fs.py` line 291 at r1](https://reviewable.io/reviews/gramineproject/gramine/1306#-NTtQ2Ia9Zj43FcITAhg:-NTtTk3iB6zmjnZVchzy:b-896fix) ([raw file](https://github.com/gramineproject/gramine/blob/c91546a4e60fe97d50cd455601f44d637be6e186/libos/test/fs/test_fs.py#L291)):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

`any workload that uses it`
</blockquote></details>

Done.
___
*[`libos/test/fs/test_fs.py` line 296 at r1](https://reviewable.io/reviews/gramineproject/gramine/1306#-NTtQA-iD725q471G5dI:-NTtTkLz8h10pS-cdiOr:b-896fix) ([raw file](https://github.com/gramineproject/gramine/blob/c91546a4e60fe97d50cd455601f44d637be6e186/libos/test/fs/test_fs.py#L296)):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

ditto
</blockquote></details>

Done.
___
*[`libos/test/fs/test_fs.py` line 300 at r1](https://reviewable.io/reviews/gramineproject/gramine/1306#-NTtQCCwEbG6S9q_kzof:-NTtTl8UChzV03xAFKhA:b-896fix) ([raw file](https://github.com/gramineproject/gramine/blob/c91546a4e60fe97d50cd455601f44d637be6e186/libos/test/fs/test_fs.py#L300)):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

ditto
</blockquote></details>

Done.

**Review by oshogbo — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/1306)* status: 0 of 1 files reviewed, 7 unresolved discussions, not enough approvals from maintainers (3 more required), not enough approvals from different teams (2 more required, approved so far: ), "squash! " found in commit messages' one-liners (waiting on @dimakuv)

___
*[`libos/test/fs/test_fs.py` line 292 at r1](https://reviewable.io/reviews/gramineproject/gramine/1306#-NTtQ5y475m8oDanIeVy:-NTtUQ3OBAWqQAZI2dwd:b-896fix) ([raw file](https://github.com/gramineproject/gramine/blob/c91546a4e60fe97d50cd455601f44d637be6e186/libos/test/fs/test_fs.py#L292)):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

`...not implemented in Linux-SGX PAL`
</blockquote></details>

Done.

**Review by dimakuv — CHANGES_REQUESTED** [review]
> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/1306)* status: all files reviewed, 1 unresolved discussion, not enough approvals from maintainers (3 more required), not enough approvals from different teams (2 more required, approved so far: ), "squash! " found in commit messages' one-liners (waiting on @oshogbo)

___
*[`libos/test/fs/test_fs.py` line 291 at r1](https://reviewable.io/reviews/gramineproject/gramine/1306#-NTtQ2Ia9Zj43FcITAhg:-NTtUbgSCZ4of-mZdyKy:bfhqmln) ([raw file](https://github.com/gramineproject/gramine/blob/c91546a4e60fe97d50cd455601f44d637be6e186/libos/test/fs/test_fs.py#L291)):*
<details><summary><i>Previously, oshogbo (Mariusz Zaborski) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Not done?

**Review by dimakuv — APPROVED** [review]
> Reviewed 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/1306)* status: all files reviewed, all discussions resolved, not enough approvals from maintainers (2 more required), not enough approvals from different teams (1 more required, approved so far: Intel), "fixup! " found in commit messages' one-liners

**Review by mkow — CHANGES_REQUESTED** [review]
> Reviewed 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/1306)* status: all files reviewed, 2 unresolved discussions, not enough approvals from maintainers (2 more required), not enough approvals from different teams (1 more required, approved so far: Intel), "fixup! " found in commit messages' one-liners (waiting on @oshogbo)

___
*[`libos/test/fs/test_fs.py` line 291 at r3](https://reviewable.io/reviews/gramineproject/gramine/1306#-NTuzqn8Esk3_YmEa2-Y:-NTuzqn8Esk3_YmEa2-Z:boygnjd) ([raw file](https://github.com/gramineproject/gramine/blob/9765c2ab134b4f0c7824197fc8c892a892bf09cf/libos/test/fs/test_fs.py#L291)):*
> ```Python
>     # These tests are disabled for Linux-SGX because it doesn't implement mmap(PROTO_WRITE)
>     # which is used in this test. We decided not to implement mmap(PROTO_WRITE) as we don't
>     # know any workload that uses it.
> ```

I don't understand this description at all. Since when we don't implement ordinary, writable memory mappings? And no workload is using writable memory? Something is wrong in this description.
___
*[`libos/test/fs/test_fs.py` line 292 at r3](https://reviewable.io/reviews/gramineproject/gramine/1306#-NTuyUx-4eT-FtyYTOxD:-NTuyUx-4eT-FtyYTOxE:b-7ywogm) ([raw file](https://github.com/gramineproject/gramine/blob/9765c2ab134b4f0c7824197fc8c892a892bf09cf/libos/test/fs/test_fs.py#L292)):*
> ```Python
>     # which is used in this test. We decided not to implement mmap(PROTO_WRITE) as we don't
>     # know any workload that uses it.
>     @unittest.skipIf(HAS_SGX, 'mmap(PROTO_WRITE) not implemented in Linux-SGX PAL')
> ```

ditto below and above



_Suggestion:_
```Python
PROT_WRITE
```

**Review by kailun-qin — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/1306)* status: all files reviewed, 3 unresolved discussions, not enough approvals from maintainers (2 more required), not enough approvals from different teams (1 more required, approved so far: Intel), "fixup! " found in commit messages' one-liners (waiting on @mkow and @oshogbo)

___
*[`libos/test/fs/test_fs.py` line 290 at r2](https://reviewable.io/reviews/gramineproject/gramine/1306#-NTvNGTUEF39Zm5kqQaq:-NTvNGTUEF39Zm5kqQar:b-4gcr8p) ([raw file](https://github.com/gramineproject/gramine/blob/3b0aab4874f70494ad5da7e62f36597a7ef6efee/libos/test/fs/test_fs.py#L290)):*
> ```Python
> 
>     # These tests are disabled for Linux-SGX because it doesn't implement mmap(PROTO_WRITE)
>     # which is used in this test. We decided not to implement mmap(PROTO_WRITE) as we don't
> ```

plural form (to keep aligned w/ the subject)?

I'd simply rephrase this to sth. like "these tests require `mmap(PROT_WRITE)` which is currently unsupported in Linux-SGX PAL, skipping them".



_Code quote:_
```Python
in this test
```
___
*[`libos/test/fs/test_fs.py` line 291 at r3](https://reviewable.io/reviews/gramineproject/gramine/1306#-NTuzqn8Esk3_YmEa2-Y:-NTvOVv405oZd2dYbIe3:b32v57) ([raw file](https://github.com/gramineproject/gramine/blob/9765c2ab134b4f0c7824197fc8c892a892bf09cf/libos/test/fs/test_fs.py#L291)):*
<details><summary><i>Previously, mkow (Michał Kowalczyk) wrote…</i></summary><blockquote>

I don't understand this description at all. Since when we don't implement ordinary, writable memory mappings? And no workload is using writable memory? Something is wrong in this description.
</blockquote></details>

+1

**Review by oshogbo — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/1306)* status: 0 of 1 files reviewed, 5 unresolved discussions, not enough approvals from maintainers (3 more required), not enough approvals from different teams (2 more required, approved so far: ), "fixup! " found in commit messages' one-liners (waiting on @dimakuv, @kailun-qin, and @mkow)

___
*[`libos/test/fs/test_fs.py` line 290 at r2](https://reviewable.io/reviews/gramineproject/gramine/1306#-NTvNGTUEF39Zm5kqQaq:-NTwbdcg6QIcMaoZR3Eh:b-17q4fl) ([raw file](https://github.com/gramineproject/gramine/blob/3b0aab4874f70494ad5da7e62f36597a7ef6efee/libos/test/fs/test_fs.py#L290)):*
<details><summary><i>Previously, kailun-qin (Kailun Qin) wrote…</i></summary><blockquote>

plural form (to keep aligned w/ the subject)?

I'd simply rephrase this to sth. like "these tests require `mmap(PROT_WRITE)` which is currently unsupported in Linux-SGX PAL, skipping them".
</blockquote></details>

Rewritten.
___
*[`libos/test/fs/test_fs.py` line 289 at r3](https://reviewable.io/reviews/gramineproject/gramine/1306#-NTwZm8yDMMglNsx2t2E:-NTwbX7dD5TCOj9nS7g_:bt2vr1b) ([raw file](https://github.com/gramineproject/gramine/blob/9765c2ab134b4f0c7824197fc8c892a892bf09cf/libos/test/fs/test_fs.py#L289)):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

You have a typo: should be `PROT_WRITE` (not `PROTO_...`)

Please fix everywhere.
</blockquote></details>

Fixed.
___
*[`libos/test/fs/test_fs.py` line 289 at r3](https://reviewable.io/reviews/gramineproject/gramine/1306#-NTwZpny6noKuUprUkdO:-NTwbYp_5GdTWxFZMCN6:bp96bek) ([raw file](https://github.com/gramineproject/gramine/blob/9765c2ab134b4f0c7824197fc8c892a892bf09cf/libos/test/fs/test_fs.py#L289)):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

I guess it's unclear to reviewers, so we should make it very explicit: `it doesn't implement mmap(PROT_WRITE, MAP_SHARED, /*fd=*/file_fd) combination of flags (i.e., shared writable file-backed memory)`.
</blockquote></details>

I have rewritten the whole part. Please check if its better.
___
*[`libos/test/fs/test_fs.py` line 291 at r3](https://reviewable.io/reviews/gramineproject/gramine/1306#-NTuzqn8Esk3_YmEa2-Y:-NTwbfVA9e7Deah-9EGF:b-7vkd1) ([raw file](https://github.com/gramineproject/gramine/blob/9765c2ab134b4f0c7824197fc8c892a892bf09cf/libos/test/fs/test_fs.py#L291)):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

> Since when we don't implement ordinary, writable memory mappings?

We don't implement **shared writable file-backed** memory mappings on Linux-SGX. See https://github.com/gramineproject/gramine/blob/032e35ecf32096cc3e8134f90dadad770b545fe0/pal/src/host/linux-sgx/pal_files.c#L261-L268

> And no workload is using writable memory?

Yes, as far as we know. Again, we're talking only about **shared writable file-backed** memory.
</blockquote></details>

I have rewritten the whole part.
___
*[`libos/test/fs/test_fs.py` line 292 at r3](https://reviewable.io/reviews/gramineproject/gramine/1306#-NTuyUx-4eT-FtyYTOxD:-NTwXjWI9HMsYLanWKZ4:b-896fix) ([raw file](https://github.com/gramineproject/gramine/blob/9765c2ab134b4f0c7824197fc8c892a892bf09cf/libos/test/fs/test_fs.py#L292)):*
<details><summary><i>Previously, mkow (Michał Kowalczyk) wrote…</i></summary><blockquote>

ditto below and above
</blockquote></details>

Done.

**Review by dimakuv — APPROVED** [review]
> Reviewed 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/1306)* status: all files reviewed, 3 unresolved discussions, not enough approvals from maintainers (2 more required), not enough approvals from different teams (1 more required, approved so far: Intel), "fixup! " found in commit messages' one-liners (waiting on @kailun-qin and @mkow)

___
*[`libos/test/fs/test_fs.py` line 289 at r3](https://reviewable.io/reviews/gramineproject/gramine/1306#-NTwZpny6noKuUprUkdO:-NTwghHS8vKqzOXgCvtS:b6vsp4v) ([raw file](https://github.com/gramineproject/gramine/blob/9765c2ab134b4f0c7824197fc8c892a892bf09cf/libos/test/fs/test_fs.py#L289)):*
<details><summary><i>Previously, oshogbo (Mariusz Zaborski) wrote…</i></summary><blockquote>

I have rewritten the whole part. Please check if its better.
</blockquote></details>

A bit too long for my taste, but I'm not blocking. It's ok to me.

**Review by kailun-qin — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/1306)* status: all files reviewed, 3 unresolved discussions, not enough approvals from maintainers (2 more required), not enough approvals from different teams (1 more required, approved so far: Intel), "fixup! " found in commit messages' one-liners (waiting on @mkow and @oshogbo)

___
*[`libos/test/fs/test_fs.py` line 290 at r4](https://reviewable.io/reviews/gramineproject/gramine/1306#-NTwkure6AbO9Kq0JE0P:-NTwkure6AbO9Kq0JE0Q:b-hz66xj) ([raw file](https://github.com/gramineproject/gramine/blob/dac426a7cc875a86077875362d921bedd1ecf905/libos/test/fs/test_fs.py#L290)):*
> ```Python
> 
>     # Gramine's implementation of file_map doesn't currently support shared memory-mapped
>     # files with write permission in PAL/Linux-SGX (like mmap(PROT_WRITE, MAP_SHARED, fd)).
> ```

Not wrapped to 100 chars?



_Code quote:_
```Python
    # Gramine's implementation of file_map doesn't currently support shared memory-mapped
    # files with write permission in PAL/Linux-SGX (like mmap(PROT_WRITE, MAP_SHARED, fd)).
```

**Review by oshogbo — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/1306)* status: all files reviewed, 3 unresolved discussions, not enough approvals from maintainers (2 more required), not enough approvals from different teams (1 more required, approved so far: Intel), "fixup! " found in commit messages' one-liners (waiting on @kailun-qin and @mkow)

___
*[`libos/test/fs/test_fs.py` line 290 at r4](https://reviewable.io/reviews/gramineproject/gramine/1306#-NTwkure6AbO9Kq0JE0P:-NTwnyG78TyC38JcYwo8:b-meqtad) ([raw file](https://github.com/gramineproject/gramine/blob/dac426a7cc875a86077875362d921bedd1ecf905/libos/test/fs/test_fs.py#L290)):*
<details><summary><i>Previously, kailun-qin (Kailun Qin) wrote…</i></summary><blockquote>

Not wrapped to 100 chars?
</blockquote></details>

Yes, we could move the files, line up but TBH I prefer this split of sentence.
If you have a strong opinion about it I can change it.

**Review by kailun-qin — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/1306)* status: all files reviewed, 2 unresolved discussions, not enough approvals from maintainers (1 more required), not enough approvals from different teams (1 more required, approved so far: Intel), "fixup! " found in commit messages' one-liners (waiting on @mkow)

___
*[`libos/test/fs/test_fs.py` line 290 at r4](https://reviewable.io/reviews/gramineproject/gramine/1306#-NTwkure6AbO9Kq0JE0P:-NTwq3jKBJkr75fVn1mr:b9z9q2x) ([raw file](https://github.com/gramineproject/gramine/blob/dac426a7cc875a86077875362d921bedd1ecf905/libos/test/fs/test_fs.py#L290)):*
<details><summary><i>Previously, oshogbo (Mariusz Zaborski) wrote…</i></summary><blockquote>

Yes, we could move the files, line up but TBH I prefer this split of sentence.
If you have a strong opinion about it I can change it.
</blockquote></details>

Gotcha. I'm fine w/ it.

**Review by mkow — CHANGES_REQUESTED** [review]
> Reviewed 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/1306)* status: all files reviewed, 2 unresolved discussions, not enough approvals from maintainers (1 more required), not enough approvals from different teams (1 more required, approved so far: Intel), "fixup! " found in commit messages' one-liners (waiting on @oshogbo)

___
*[`libos/test/fs/test_fs.py` line 291 at r3](https://reviewable.io/reviews/gramineproject/gramine/1306#-NTuzqn8Esk3_YmEa2-Y:-NTxVckI-kpRt7Es-Q0B:bdpfvya) ([raw file](https://github.com/gramineproject/gramine/blob/9765c2ab134b4f0c7824197fc8c892a892bf09cf/libos/test/fs/test_fs.py#L291)):*
> We don't implement **shared writable file-backed** memory mappings on Linux-SGX. See 

Ok, that I know, but the description nowhere mentioned that it's about file mappings nor shareable mappings ;)
___
*[`libos/test/fs/test_fs.py` line 291 at r4](https://reviewable.io/reviews/gramineproject/gramine/1306#-NTxW2fA2K6krVw4dGTL:-NTxW2fA2K6krVw4dGTM:undefined) ([raw file](https://github.com/gramineproject/gramine/blob/dac426a7cc875a86077875362d921bedd1ecf905/libos/test/fs/test_fs.py#L291)):*
> ```Python
>     # Gramine's implementation of file_map doesn't currently support shared memory-mapped
>     # files with write permission in PAL/Linux-SGX (like mmap(PROT_WRITE, MAP_SHARED, fd)).
>     # These tests require it because of that we skip them. We decided not to implement it as we
> ```





_Suggestion:_
```Python
These tests require it, so we skip them.
```
___
*[`libos/test/fs/test_fs.py` line 292 at r4](https://reviewable.io/reviews/gramineproject/gramine/1306#-NTxW8N2B1-fWU2dSe1e:-NTxW8N2B1-fWU2dSe1f:undefined) ([raw file](https://github.com/gramineproject/gramine/blob/dac426a7cc875a86077875362d921bedd1ecf905/libos/test/fs/test_fs.py#L292)):*
> ```Python
>     # files with write permission in PAL/Linux-SGX (like mmap(PROT_WRITE, MAP_SHARED, fd)).
>     # These tests require it because of that we skip them. We decided not to implement it as we
>     # don't know any workload with it.
> ```





_Suggestion:_
```Python
any workload using it.
```

**Review by oshogbo — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/1306)* status: all files reviewed, 2 unresolved discussions, not enough approvals from maintainers (3 more required), not enough approvals from different teams (2 more required, approved so far: ), "fixup! " found in commit messages' one-liners (waiting on @mkow)

___
*[`libos/test/fs/test_fs.py` line 291 at r4](https://reviewable.io/reviews/gramineproject/gramine/1306#-NTxW2fA2K6krVw4dGTL:-NU0jWyY6eD_Fs_kJ8hV:b-896fix) ([raw file](https://github.com/gramineproject/gramine/blob/dac426a7cc875a86077875362d921bedd1ecf905/libos/test/fs/test_fs.py#L291)):*
> ```Python
>     # Gramine's implementation of file_map doesn't currently support shared memory-mapped
>     # files with write permission in PAL/Linux-SGX (like mmap(PROT_WRITE, MAP_SHARED, fd)).
>     # These tests require it because of that we skip them. We decided not to implement it as we
> ```

Done.
___
*[`libos/test/fs/test_fs.py` line 292 at r4](https://reviewable.io/reviews/gramineproject/gramine/1306#-NTxW8N2B1-fWU2dSe1e:-NU0jaIg6iJHZvSz5MTO:b-896fix) ([raw file](https://github.com/gramineproject/gramine/blob/dac426a7cc875a86077875362d921bedd1ecf905/libos/test/fs/test_fs.py#L292)):*
> ```Python
>     # files with write permission in PAL/Linux-SGX (like mmap(PROT_WRITE, MAP_SHARED, fd)).
>     # These tests require it because of that we skip them. We decided not to implement it as we
>     # don't know any workload with it.
> ```

Done.

**Review by kailun-qin — DISMISSED** [review]
> Reviewed 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/1306)* status: all files reviewed, 2 unresolved discussions, not enough approvals from maintainers (2 more required), not enough approvals from different teams (1 more required, approved so far: Intel), "fixup! " found in commit messages' one-liners (waiting on @mkow)

**Review by dimakuv — APPROVED** [review]
> Reviewed 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/1306)* status: all files reviewed, 2 unresolved discussions, not enough approvals from maintainers (1 more required), not enough approvals from different teams (1 more required, approved so far: Intel), "fixup! " found in commit messages' one-liners (waiting on @mkow)

**Review by mkow — DISMISSED** [review]
> Reviewed 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/1306)* status: all files reviewed, all discussions resolved, "fixup! " found in commit messages' one-liners

___
*[`libos/test/fs/test_fs.py` line 291 at r4](https://reviewable.io/reviews/gramineproject/gramine/1306#-NTxW2fA2K6krVw4dGTL:-NU1WiTdDz2XHyME-oNY:b-k0kt0) ([raw file](https://github.com/gramineproject/gramine/blob/dac426a7cc875a86077875362d921bedd1ecf905/libos/test/fs/test_fs.py#L291)):*
<details><summary><i>Previously, oshogbo (Mariusz Zaborski) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Not blocking, but sounds weird to me without "we".

**Review by mkow — APPROVED** [review]
> Reviewed 1 of 1 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/1306)* status: all files reviewed, all discussions resolved, not enough approvals from maintainers (1 more required), not enough approvals from different teams (1 more required, approved so far: ITL)

**Review by dimakuv — APPROVED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/1306)* status: :shipit: complete! all files reviewed, all discussions resolved
